control/content/index.html

```diff
@@ -46,202 +54,186 @@ <h4 class="mdc-layout-grid__cell--span-4">Use user current location</h4>
             let address = {
-                lat: position.coords.latitude,
-                lng: position.coords.longitude
+                lat: crd.latitude,
+                lng: crd.longitude
             }
             settingsData.place.address = Object.assign(address);
             getUserAddressBy(address.lat, address.lng);
-        });
-    }
-
-    getUserAddressBy = (lat, long) => {
-        var xhttp = new XMLHttpRequest();
-        xhttp.onreadystatechange = function () {
-            if (this.readyState == 4 && this.status == 200) {
-                let address = JSON.parse(this.responseText);
-                document.getElementById('searchTextField').value = address.results[0].formatted_address || '';
-                settingsData.place.title = address.results[0].formatted_address;
-            }
-        };
-        xhttp.open("GET", "https://maps.googleapis.com/maps/api/geocode/json?latlng="+lat+","+long+"&key=AIzaSyC4Dw4EzKeyVBXWBsbO9-UgyEARL6WLrlU" , true);
-        xhttp.send();
-    }
+        }
 
+        function error(err) {
+            console.warn(`ERROR(${err.code}): ${err.message}`);
+        }
 
-    handleToggleButton = () => {
-        let toggleBtn = document.querySelector('.mdc-switch');
-        let textField = document.querySelector('.city__field');
-        let inputField = document.querySelector('.city__input');
+        getUserCurrentPosition = () => {
+            navigator.geolocation.getCurrentPosition(success, error, options);
+            /*Settings.getGeoPosition((err, position) => {
+                if (err) console.log(err);
+                let address = {
+                    lat: position.coords.latitude,
+                    lng: position.coords.longitude
+                }
+                settingsData.place.address = Object.assign(address);
+                getUserAddressBy(address.lat, address.lng);
+            });*/
+        }
 
-        if(toggleBtn.classList.contains('mdc-switch--checked')) {
-            toggleBtn.classList.remove("mdc-switch--checked");
-            textField.classList.remove('mdc-text-field--disabled');
-            inputField.classList.remove("disabled");
-            settingsData.useUserCurrentLocation = false;
+        getUserAddressBy = (lat, long) => {
+            var xhttp = new XMLHttpRequest();
+            xhttp.onreadystatechange = function () {
+                if (this.readyState == 4 && this.status == 200) {
+                    let address = JSON.parse(this.responseText);
+                    document.getElementById('searchTextField').value = address.results[0].formatted_address || '';
+                    settingsData.place.title = address.results[0].formatted_address;
+                    handleSave();
+                }
+            };
+            xhttp.open("GET", "https://maps.googleapis.com/maps/api/geocode/json?latlng=" + lat + "," + long + "&key=AIzaSyC4Dw4EzKeyVBXWBsbO9-UgyEARL6WLrlU", true);
```

Comment:
the API key field should not be hardcoded. The api key value should be pulled from the sdk. https://github.com/BuildFire/sdk/wiki/API-Keys. 

You can get the APIKEY from buildfire.getContext function